CMakeLists.txt

```diff
@@ -55,11 +53,31 @@ option(DLAF_BUILD_DOC "Build documentation" OFF)
 option(DLAF_WITH_PRECOMPILED_HEADERS "Use precompiled headers." OFF)
 option(DLAF_WITH_SCALAPACK "Build ScaLAPACK-like C API (requires ScaLAPACK)" OFF)
 
-if(DLAF_WITH_MKL AND DLAF_WITH_MKL_LEGACY)
-  message(SEND_ERROR "DLAF_WITH_MKL and DLAF_WITH_MKL_LEGACY are mutually exclusive.")
+option(DLAF_WITH_MKL "Enable MKL as provider for BLAS and LAPACK" "${DLAF_WITH_MKL_TRY_COMPILE}")
```

Comment:
since `DLAF_WITH_MKL_TRY_COMPILE` will be set later, doesn't this mean that by default `DLAF_WITH_MKL=""`?

Reply:
Hrm, excellent catch. This should not be an option at all, otherwise it won't be undefined if a user doesn't set it. Let me fix that.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1149/commits/9dc2f78d8ff707e8d6c64f517f5929bc83dcdf63